repos/system_upgrade/el7toel8/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -77,13 +77,10 @@ def get_system_groups_status():
 
 @aslist
 def _get_active_kernel_modules(logger):
-    lines = run(['lsmod'], split=True)['stdout']
-    for l in lines[1:]:
-        name = l.split(' ')[0]
+    lines = [module for module in os.listdir("/sys/module") if os.path.isdir("/sys/module/{}".format(module))]
 
-        # Read parameters of the given module as exposed by the
-        # `/sys` VFS, if there are no parameters exposed we just
-        # take the name of the module
+    for l in lines:
+        name = l
```

Comment:
- We should always use `os.path.join` instead of format string as was used originally already. We missed that earlier as I see.
- nitpick - when it is possible use for strings same literal as it is used in the rest of code / file. In this case prefer apostrophe instead of quotes
- It is not good idea to use variable names like `l`. One of problems is, that it can be simply interchanged with `I` or `1` (depends on used font). Additionally, it is confusing to get information what the variable represent. In this case, the rename on the another line gives you hint that you should do that on one line of code.
- You do not need to use `lines` name anymore as now it is not output of subcommand, but you already provide information what modules exist on the system.

What about something like:
```python
    modules = [module for module in os.listdir('/sys/module') if os.path.isdir(os.path.join('/sys/module/', module))]
    for module in modules:
        base_path = os.path.join('/sys/module/', module)
```

Reply:
@Seenivasanseeni, could you please take a look?